src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1613,6 +1613,10 @@ internal virtual FieldSymbol FixedElementField
             }
         }
 
+#nullable enable
+        internal abstract bool HasCollectionBuilderAttribute(out TypeSymbol? builderType, out string? methodName);
```

Comment:
nit: and i can see it both ways.  but it seems like just making this virtual, and updating the single place that actually overrides, would be cleaner.  i think this primarily because there isn't an expectation of ever having this attribute somehow come in on any other sort of type (since... well, it's an attribute).

Reply:
welp.  i was wrong.  ignore me :D

Reply:
For these methods we often have multiple implementations (often distinct implementations for source, PE, substituted, and retargeting symbols, as is the case here), and the common pattern is to use an `abstract` base method (rather than `virtual`) to avoid assuming the base implementation is correct for existing (and future) symbols.